components/experimental/src/units/converter_factory.rs

```diff
@@ -43,29 +42,17 @@ impl ConverterFactory {
     icu_provider::gen_any_buffer_data_constructors!(
         locale: skip,
         options: skip,
-        error: ConversionError,
-        #[cfg(skip)]
-    );
+        error: DataError,
 
-    /// Creates a new [`ConverterFactory`] from compiled locale data and an options bag.
+    /// Creates a new [`ConverterFactory`] from compiled locale data.
     ///
     /// ✨ *Enabled with the `compiled_data` Cargo feature.*
     ///
     /// [📚 Help choosing a constructor](icu_provider::constructors)
-    #[cfg(feature = "compiled_data")]
-    pub fn try_new() -> Result<Self, ConversionError> {
-        let payload = crate::provider::Baked
-            .load(DataRequest {
-                locale: &DataLocale::default(),
-                metadata: Default::default(),
-            })?
-            .take_payload()?;
-
-        Ok(Self { payload })
-    }
+    );
 
     #[doc = icu_provider::gen_any_buffer_unstable_docs!(UNSTABLE, Self::try_new)]
-    pub fn try_new_unstable<D>(provider: &D) -> Result<Self, ConversionError>
+    pub fn try_new_unstable<D>(provider: &D) -> Result<Self, DataError>
```

Comment:
ok, this function takes only a `provider` argument so this is consistent with https://github.com/unicode-org/icu4x/pull/4638#issuecomment-1994218244